### DISCUSSION TITLE
Accessing Node Attributes from Graph in Python

### DISCUSSION THREAD

**Discussion by falahamro** [networkit/networkit#1008] [Q&A]
I am reading a gephi file (.graphml) into Networkit. While I am calcualting the group centrality measures, I am able to identify the nodes represented as intigers in python. How can I find the metadata assicated with those nodes? How do I map the input file nodes with the intigers? 

Are those intigers I am getting match with order of the nodes in the input file? 

Thank you.

**Comment by fabratu** [comment]
Hi,

the GraphML-reader does currently ignores node metadata. Also there is unfortunately no support for returning the node mapping. Internally it is used to create the correct edges, however this is not available via API. The reader works on every line from top to bottom, so the first found node has ID 0 and so on.

Besides the reader: In case you want to use general node attributes, you first have to create/attach them. After that you can access them via the node ID.
```
# Create a new attribute named 'myAtt' of type 'str'
att = G.attachNodeAttribute("myAtt", str)

# Set attribute values
att[0] = "foo" # Attribute of node 0
att[1] = "bar" # Attribute of node 1
```
Also see here for more examples: https://networkit.github.io/dev-docs/notebooks/GraphNotebook.html#Node-Attributes

**Reply by falahamro** [reply]
Thank you; it would be a great feature to have. In the meantime, I managed to map the node intigers to the original nodes by ready the same gephi in a different format and calling the node numbers.
